/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax = "proto3";

import "WifiDirectConfigData.proto";

package oaa.proto.messages;

// Wire format verified against WirelessAndroidAutoDongle (nisargjhaveri).
// Field numbers MUST match what the phone expects â€” the original aasdk had
// them wrong (key=1, security_mode=2, ssid=3, no bssid).
// Correct order: ssid=1, key=2, bssid=3, security_mode=4, access_point_type=5.
message WifiSecurityResponse {
    optional string ssid = 1;
    optional string key = 2;
    optional string bssid = 3;
    optional SecurityMode security_mode = 4;
    optional AccessPointType access_point_type = 5;

    // Fields 6-11 discovered via APK proto decoder (wan.java, AA v16.1)
    optional data.WifiDirectConfig wifi_direct_config = 6;  // WiFi Direct connection params (APK: wbb)
    optional string ip_address = 7;         // HU IP address for direct connection
    optional string gateway = 8;            // Gateway IP
    optional uint32 prefix_length = 9;      // Network prefix length (e.g. 24)
    optional bool hidden_network = 10;      // Whether AP is hidden (no SSID broadcast)
    optional bool band_5ghz = 11;           // Whether to use 5GHz band
    optional string unknown_12 = 12;        // Discovered via APK proto decoder (wan.java)
    optional bool unknown_13 = 13;          // Discovered via APK proto decoder (wan.java)

    enum SecurityMode {
        UNKNOWN_SECURITY_MODE = 0;
        OPEN = 1;
        WEP_64 = 2;
        WEP_128 = 3;
        WPA_PERSONAL = 4;
        WPA2_PERSONAL = 8;
        WPA_WPA2_PERSONAL = 12;
        WPA_ENTERPRISE = 20;
        WPA2_ENTERPRISE = 24;
        WPA_WPA2_ENTERPRISE = 28;
    }

    enum AccessPointType {
        STATIC = 0;
        DYNAMIC = 1;
    }
}
